manifests/init.pp

```diff
@@ -78,34 +115,112 @@
 # Copyright 2014-2015 Gjermund Jensvoll
 #
 class colorprompt (
-  $ensure            = $colorprompt::params::ensure,
-  $path              = $colorprompt::params::path,
-  $default_usercolor = $colorprompt::params::default_usercolor,
-  $custom_usercolors = $colorprompt::params::custom_usercolors,
-  $server_color      = $colorprompt::params::server_color,
-  $env_name          = $colorprompt::params::env_name,
-  $env_color         = $colorprompt::params::env_color,
-  $prompt            = $colorprompt::params::prompt,
-  $modify_skel       = $colorprompt::params::modify_skel,
-  $modify_root       = $colorprompt::params::modify_root,
+  $ensure                  = $colorprompt::params::ensure,
+  $path                    = $colorprompt::params::path,
+  $default_usercolor       = $colorprompt::params::default_usercolor,
+  $custom_usercolors       = $colorprompt::params::custom_usercolors,
+  $server_color            = $colorprompt::params::server_color,
+  $pwd_color               = $colorprompt::params::pwd_color,
+  $env_name                = $colorprompt::params::env_name,
+  $env_color               = $colorprompt::params::env_color,
+  $git_prompt              = $colorprompt::params::git_prompt,
+  $git_prompt_path         = $colorprompt::params::git_prompt_path,
+  $prompt                  = $colorprompt::params::prompt,
+  $modify_skel             = $colorprompt::params::modify_skel,
+  $modify_root             = $colorprompt::params::modify_root,
+  $bash_completion_package = $colorprompt::params::bash_completion_package,
+  $git_package             = $colorprompt::params::git_package,
+  $fg_colors            = $colorprompt::params::fg_colors,
+  $bg_colors           = $colorprompt::params::bg_colors,
+  $styles            = $colorprompt::params::styles,
 ) inherits colorprompt::params {
 
   validate_re($ensure, '^(present|absent)$',
     "${ensure} is not supported for ensure. Allowed values are 'present' and 'absent'.")
 
   validate_absolute_path(
     $path,
+    $git_prompt_path,
   )
 
   validate_string(
-    $prompt
+    $prompt,
+    $bash_completion_package,
+    $git_package,
   )
 
   validate_bool(
+    $git_prompt,
     $modify_skel,
-    $modify_root
+    $modify_root,
+  )
+
+  validate_hash(
+    $custom_usercolors,
+    $fg_colors,
+    $bg_colors,
+    $styles,
   )
 
+  if empty($custom_usercolors) {
+    fail('Value of parameter $custom_usercolors must be not empty!')
+  } else {
+    if ( versioncmp($::puppetversion, '4.0.0') >= 0 ) {
+      $custom_usercolors.each | $user, $colors | {
+        validate_colors($colors, "Values of \$custom_usercolors not correct for user '${user}'! Value")
+      }
+    }
+  }
+
+  unless is_string($default_usercolor) or is_array($default_usercolor) {
+    fail('Value of parameter $default_usercolor must be string or array!')
+  } else {
```

Comment:
'Unless' statements in puppet version 3.x does not support else clause. It'll cause a syntax error.